OpenRA.Mods.Common/UtilityCommands/ExtractSequenceDocsCommand.cs

```diff
@@ -0,0 +1,98 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Newtonsoft.Json;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Primitives;
+
+namespace OpenRA.Mods.Common.UtilityCommands
+{
+	class ExtractSequenceDocsCommand : IUtilityCommand
+	{
+		string IUtilityCommand.Name => "--sequence-docs";
+
+		bool IUtilityCommand.ValidateArguments(string[] args)
+		{
+			return true;
+		}
+
+		[Desc("[VERSION]", "Generate sprite sequence documentation in JSON format.")]
+		void IUtilityCommand.Run(Utility utility, string[] args)
+		{
+			// HACK: The engine code assumes that Game.modData is set.
+			Game.ModData = utility.ModData;
+
+			var version = utility.ModData.Manifest.Metadata.Version;
+			if (args.Length > 1)
+				version = args[1];
+
+			var objectCreator = utility.ModData.ObjectCreator;
+			var traitInfos = objectCreator.GetTypesImplementing<ISpriteSequence>().OrderBy(t => t.Namespace);
+
+			var json = GenerateJson(version, traitInfos);
+			Console.WriteLine(json);
+		}
+
+		static string GenerateJson(string version, IEnumerable<Type> sequenceTypes)
+		{
+			var sequenceTypesInfo = sequenceTypes.Where(x => !x.ContainsGenericParameters && !x.IsAbstract)
+				.Where(x => x.Name != nameof(FileNotFoundSequence)) // HACK: This is the simplest way to exclude FileNotFoundSequence, which shouldn't be added.
```

Comment:
Not really a hack though.

Reply:
I'll change to "NOTE:" I guess. I think it is still worth noting.

Reply:
The `FileNotFoundSequence` is the hack I guess.